generator/base_files/BaseTexFile.py

```diff
@@ -97,45 +109,84 @@ def sort_class_names(self, classes):
     @staticmethod
     def sort_attribute_names(classes):
+        """
+        Function to create texnames for attributes.
+
+        :param classes:
+        """
         if classes is not None:
             for i in range(0, len(classes)):
                 attribs = classes[i]['attribs']
-                for j in range(0, len(attribs)):
-                    if attribs[j]['type'] == 'lo_element' or attribs[j]['type'] == 'element' or attribs[j]['type'] == 'inline_lo_element':
-                        name = attribs[j]['element']
-                    else:
-                        name = attribs[j]['name']
-                    texname = strFunctions.texify(name)
-                    attribs[j]['texname'] = texname
+                BaseTexFile.update_attrib_dicts(attribs)
+                # for j in range(0, len(attribs)):
+                #     if attribs[j]['type'] in ['lo_element', 'element',
+                #                               'inline_lo_element']:
+                #         name = attribs[j]['element']
+                #     else:
+                #         name = attribs[j]['name']
+                #     texname = strFunctions.texify(name)
+                #     attribs[j]['texname'] = texname
             for i in range(0, len(classes)):
                 if 'lo_attribs' in classes[i]:
                     lo_attribs = classes[i]['lo_attribs']
-                    for j in range(0, len(lo_attribs)):
-                        if lo_attribs[j]['type'] == 'lo_element' or lo_attribs[j]['type'] == 'element' or lo_attribs[j]['type'] == 'inline_lo_element' :
-                            name = lo_attribs[j]['element']
-                        else:
-                            name = lo_attribs[j]['name']
-                        texname = strFunctions.texify(name)
-                        lo_attribs[j]['texname'] = texname
-
-    # function to create texnames for enums
+                    BaseTexFile.update_attrib_dicts(lo_attribs)
+                    # for j in range(0, len(lo_attribs)):
+                    #     if lo_attribs[j]['type'] in ['lo_element', 'element',
+                    #                                  'inline_lo_element']:
+                    #         name = lo_attribs[j]['element']
+                    #     else:
+                    #         name = lo_attribs[j]['name']
+                    #     texname = strFunctions.texify(name)
+                    #     lo_attribs[j]['texname'] = texname
+
+    @staticmethod
+    def update_attrib_dicts(my_list):
+        """
+        Update a list of attribute dictionaries.
+
+        :param my_list: the list which we update.
+        """
+        for j in range(0, len(my_list)):
+            if my_list[j]['type'] in ['lo_element', 'element',
+                                      'inline_lo_element']:
+                name = my_list[j]['element']
+            else:
+                name = my_list[j]['name']
+            texname = strFunctions.texify(name)
```

Comment:
not SF ? :-)

Reply:
I suppose I should have done, but IIRC it was only used once or twice in this module, so I didn't bother.